source/Camera/InfoUtilOther/ZividBenchmark/ZividBenchmark.cpp

```diff
@@ -0,0 +1,453 @@
+            filterProcessingDurations.push_back(std::get<1>(meanAndAverageFilterDurations));
+        }
+        printFilterResults(filterProcessingDurations);
+    }
+
+    Zivid::Settings2D makeSettings2D(const size_t exposureTime)
+    {
+        Zivid::Settings2D settings;
+        settings.set(Zivid::Settings2D::ExposureTime(std::chrono::microseconds{ exposureTime }));
+        return settings;
+    }
+
+    void benchmarkCapture2D(Zivid::Camera &camera, const Zivid::Settings2D &settings, const size_t numFrames)
+    {
+        printCapture2DHeader(numFrames, settings);
+
+        for(size_t i = 0; i < 5; i++) // Warmup frames
+        {
+            camera.capture2D(settings);
+        }
+
+        std::vector<Duration> captureDurations;
+        std::vector<Duration> allDurations;
+
+        for(size_t i = 0; i < numFrames; i++)
+        {
+            const auto beforeCapture = HighResClock::now();
+            const auto frame2D = camera.capture2D(settings);
+            const auto afterCapture = HighResClock::now();
+
+            captureDurations.push_back(afterCapture - beforeCapture);
+        }
+        allDurations.push_back(computeMedianDuration(captureDurations));
+        allDurations.push_back(computeAverageDuration(captureDurations));
+
+        printCapture2DResults(allDurations);
+    }
+
+    void benchmarkSave(Zivid::Camera &camera, const size_t numFrames)
+    {
+        printSaveHeader(numFrames);
+
+        auto frame = camera.capture();
+        frame.getPointCloud();
+
+        std::vector<Duration> allDurations;
+        std::vector<std::string> fileNames{ "Zivid3D.zdf", "Zivid3D.ply", "Zivid3D.pcd", "Zivid3D.xyz" };
+        for(const auto &fileName : fileNames)
+        {
+            std::vector<Duration> durationsPerFormat;
+            for(size_t j = 0; j < numFrames; j++)
+            {
+                const auto beforeSave = HighResClock::now();
+                frame.save(fileName);
+                const auto afterSave = HighResClock::now();
+
+                durationsPerFormat.push_back(afterSave - beforeSave);
+            }
+
+            allDurations.push_back(computeMedianDuration(durationsPerFormat));
+            allDurations.push_back(computeAverageDuration(durationsPerFormat));
+        }
+        printSaveResults(allDurations);
+    }
+} // namespace
+
+int main()
+{
+    try
+    {
+        Zivid::Application zivid;
+
+        auto camera = getFirstCamera(zivid);
+
+        const size_t numConnects = 10;
+        const size_t numFrames3D = 20;
+        const size_t numFrames2D = 50;
+        const size_t numFramesSave = 10;
+
+        const size_t exposureTime = getMinExposureTime(camera.modelName());
+        const std::vector<size_t> exposureTimeOneFrame{ exposureTime };
+        const std::vector<size_t> exposureTimeTwoFrames{ exposureTime, exposureTime };
+        const std::vector<size_t> exposureTimeThreeFrames{ exposureTime, exposureTime, exposureTime };
```

Comment:
always lowest exposure time?

Reply:
Yes, this program will be run on various different hardware and the results will be compiled into a table for marketing purposes - we want to show best possible performance.